pkgs/applications/misc/zathura/core/default.nix

```diff
@@ -18,6 +23,7 @@ stdenv.mkDerivation rec {
     "RSTTOMAN=${docutils}/bin/rst2man.py"
     "VERBOSE=1"
     "TPUT=${ncurses.out}/bin/tput"
+    (stdenv.lib.strings.optionalString synctexSupport "WITH_SYNCTEX=1")
```

Comment:
Two small nitpicks:

1) Better to get `lib` as another input to avoid writing `stdenv.lib` everywhere. You also don't need to specify sub-library -- just use `lib.optionalString`.
2) `makeFlags` is a list of strings -- what you are doing is passing a flag or an empty string (conditionally) as another item in that list. While that would work, a better way would be to add or not that item itself to a list:
```
makeFlags = [
    a
    b
  ] ++ lib.optional fooSupport "WITH_FOO";
```

Kudos for adding a `Support` flag!